group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -16294,6 +16299,76 @@ barTopicName, new TopicMetadata(barTopicId, barTopicName, 3)
         );
     }
 
+    @Test
+    public void testSharePartitionKeyMap() {
+        MockPartitionAssignor assignor = new MockPartitionAssignor("range");
+        assignor.prepareGroupAssignment(new GroupAssignment(Collections.emptyMap()));
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .withConfig(GroupCoordinatorConfig.CONSUMER_GROUP_ASSIGNORS_CONFIG, List.of(assignor))
+            .build();
+
+        MetadataImage image = mock(MetadataImage.class);
+        TopicsImage topicsImage = mock(TopicsImage.class);
+        TopicImage t1image = mock(TopicImage.class);
+        TopicImage t2image = mock(TopicImage.class);
+        when(topicsImage.getTopic(anyString()))
+            .thenReturn(t1image)
+            .thenReturn(t2image);
+
+        ShareGroup shareGroup = mock(ShareGroup.class);
+        when(shareGroup.subscribedTopicNames())
+            .thenReturn(Map.of(
+                    "t1", mock(SubscriptionCount.class),
+                    "t2", mock(SubscriptionCount.class)
+                )
+            );
+
+        when(shareGroup.groupId())
+            .thenReturn("share-group");
+        when(image.topics())
+            .thenReturn(topicsImage);
+        when(image.provenance())
+            .thenReturn(new MetadataProvenance(-1, -1, -1, true));
+
+        when(t1image.partitions())
+            .thenReturn(
+                Map.of(
+                    0, mock(PartitionRegistration.class),
+                    1, mock(PartitionRegistration.class)
+                )
+            );
+        Uuid t1Uuid = Uuid.randomUuid();
+        when(t1image.id()).thenReturn(t1Uuid);
+
+        when(t2image.partitions())
+            .thenReturn(
+                Map.of(
+                    0, mock(PartitionRegistration.class),
+                    1, mock(PartitionRegistration.class)
+                )
+            );
+        Uuid t2Uuid = Uuid.randomUuid();
+        when(t2image.id()).thenReturn(t2Uuid);
+
+        context.groupMetadataManager.onNewMetadataImage(image, mock(MetadataDelta.class));
+        Map<String, Map<Uuid, List<Integer>>> keyMap = context.groupMetadataManager.sharePartitionKeysMap(List.of(shareGroup));
+        assertEquals(1, keyMap.size());
+        assertEquals(2, keyMap.get("share-group").size());
+        for (Uuid topic : List.of(t1Uuid, t2Uuid)) {
+            assertEquals(2, keyMap.get("share-group").get(topic).size());
+            assertTrue(keyMap.get("share-group").get(topic).contains(0));
+            assertTrue(keyMap.get("share-group").get(topic).contains(1));
+        }
```

Comment:
We usually prefer to build the expected map and to use assertEquals(expectedMap, actualMap).